### PR TITLE
(feature) add redis streams as an output

### PR DESCRIPTION
**Pull request by maguec**
As of Redis 5.0 the streams feature has become available.

Add output support for Redis streams

### CODE REVIEW

**Comment by kaiyan-sheng** [comment]
@maguec Thanks for your PR! Signing the CLA is needed and please see https://github.com/elastic/beats/blob/master/CONTRIBUTING.md for more details.

**Comment by maguec** [comment]
@kaiyan-sheng CLA is signed

File: libbeat/outputs/redis/client.go [hunk 1]
```diff
@@ -263,9 +271,17 @@ func (c *client) publishEventsPipeline(conn redis.Conn, command string) publishF
 			}
 
 			data = append(data, okEvents[i])
-			if err := conn.Send(command, eventKey, serializedEvent); err != nil {
-				logp.Err("Failed to execute %v: %v", command, err)
-				return okEvents, err
+			if command == "XADD" {
```
Comment (kvch) [inline libbeat/outputs/redis/client.go]: What do you think about simplifying the code to this format?

```golang
var commandArgs []string
if command == "XADD" {
    commandArgs = []string{eventKey, "*", "event", serializedEvent}
} else {
    commandArgs = []string{eventKey, serializedEvent}
}
if err := conn.Send(command, commandArgs...); err != nil {
    logp.Err("Failed to execute %v: %v", command, err)
    return okEvents, err
}
```

This way `conn.Send` is not repeated.

Comment (maguec) [reply libbeat/outputs/redis/client.go]: @kvch thanks that's much cleaner.  Updated

**Comment by urso** [comment]
Hi, @maguec, @kvch, this PR is stalled for quite some time. Supporting redis streams is a nice enhancement. Unfortunately CI is broken, and this PR should be rebased on top of master. Do you plan to continue working on it?

Marking this PR as stalled for now.  It will be closed in the next weeks if there is no more progress.

**Comment by maguec** [comment]
@urso I went ahead and created a new PR https://github.com/elastic/beats/pull/16716 that handles this.  Please feel free to close

**Comment by urso** [comment]
Thank you.

**Pull request closed without merging**